lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -318,13 +318,29 @@ static bool HasReflectionInfo(ObjectFile *obj_file) {
     return false;
   };
 
+  #ifdef __linux__
```

Comment:
We shouldn't make this a compile-time decision. You could be debugging a Linux binary from a macOS machine. That's actually a very common scenario for server development on macOS.
Ideally these strings should live in a single location in a header file in the Swift compiler.

I would suggest a Swift compiler patch that adds a new file 

`include/swift/ABI/ObjectFile.h`

where you define a base class with one Mach-O and one ELF implementation that has a `getFieldMD()` or a `getSectionName(enum SectionType)` method or something similar.

Reply:
And then use that in `ReflectionContext.h` so the strings only live in a single place.

Reply:
Ok! But if it isn't a compile-time decision how would I pick the corrent sub-class?

Reply:
One way of doing it would be to add a similar virtual method to lldb::ObjectFile whose various implementations look it up in include/swift/ABI/ObjectFile.h